compiler/src/parsing/parsetree_iter.re

```diff
@@ -3,8 +3,8 @@ open Parsetree;
 type hooks = {
   enter_location: Location.t => unit,
   leave_location: Location.t => unit,
-  enter_attribute: attribute => unit,
-  leave_attribute: attribute => unit,
+  enter_attribute: (attribute, attribute_context) => unit,
+  leave_attribute: (attribute, attribute_context) => unit,
```

Comment:
I don't like this attribute_context type. Why don't we actually change the type of attributes?

Reply:
Do you mean basically adding the `attribute_context` as a field on the `attribute` type itself? If so then should it be on the `attributes` type since everything in an attribute group should have the same "context"?

Reply:
Do we need this at all? Can the attribute wellformedness check be done inside of `enter_toplevel`, `enter_expression`, etc?